vyper/ast/utils.py

```diff
@@ -117,70 +45,47 @@ def parse_to_ast(source_code: str, source_id: int = 0) -> list:
     except SyntaxError as e:
         # TODO: Ensure 1-to-1 match of source_code:reformatted_code SyntaxErrors
         raise PythonSyntaxException(e, source_code) from e
-    annotate_python_ast(py_ast, source_code, class_types)
-    asttokens.ASTTokens(source_code, tree=py_ast)
-    # Convert to Vyper AST.
-    vy_ast = parse_python_ast(
-        source_code=source_code,
-        node=py_ast,
-        source_id=source_id,
-    )
-    return vy_ast.body  # type: ignore
-
-
-@iterable_cast(list)
-def _ast_to_list(node: list) -> Generator:
-    for x in node:
-        yield ast_to_dict(x)
+    annotate_python_ast(py_ast, source_code, class_types, source_id)
 
-
-@iterable_cast(dict)
-def _ast_to_dict(node: vy_ast.VyperNode) -> Generator:
-    for f in node.get_slots():
-        if f not in DICT_AST_SKIPLIST:
-            yield (f, ast_to_dict(getattr(node, f, None)))
-    yield ('ast_type', node.__class__.__name__)
+    # Convert to Vyper AST.
+    return vy_ast.get_node(py_ast).body  # type: ignore
```

Comment:
This recursively converts `py_ast` to `vy_ast`?

Also, eventually this should return the base AST node directly, right?

Reply:
Yes, and yes. My next PR already has modified it to return `Module` instead of a list :)

Reply:
...but it would probably be better here anyway.

Reply:
Done, and tests passing. I made `Module` iterable to reduce the pain of this change.